src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -187,6 +189,15 @@ public String deleteEventType(Request request, Response response) {
                     halt(HttpStatusCodes.ERROR, "could not delete corresponding mappings of event type");
                 }
 
+                // delete mapping conditions
+                for (EventEntityMapping mapping : mappings) {
+                    List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mapping.getId());
+                    MappingCondition[] conditionArray = new MappingCondition[mappings.size()];
+                    if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(conditions.toArray(conditionArray))) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS134)